src/org/zaproxy/zap/extension/openapi/ImportFromUrlDialog.java

```diff
@@ -0,0 +1,139 @@
+
+public class ImportFromUrlDialog extends AbstractDialog implements ActionListener {
+
+    private static final long serialVersionUID = -7074394202143400215L;
+    private static final String MESSAGE_PREFIX = "openapi.importfromurldialog.";
+
+    private ExtensionOpenApi caller;
+
+    private JLabel labelURL = new JLabel(Constant.messages.getString(MESSAGE_PREFIX + "labelURL"));
+    private JTextField fieldURL = new JTextField(30);
+
+    private JButton buttonCancel = new JButton(Constant.messages.getString("all.button.cancel"));
+    private JButton buttonImport = new JButton(Constant.messages.getString(MESSAGE_PREFIX + "importButton"));
+
+    public ImportFromUrlDialog(JFrame parent, ExtensionOpenApi caller) {
+        super(parent, true);
+        this.setTitle(Constant.messages.getString(MESSAGE_PREFIX + "title"));
+        this.caller = caller;
+        if (parent != null) {
+            Dimension parentSize = parent.getSize();
+            Point p = parent.getLocation();
+            setLocation(p.x + parentSize.width / 4, p.y + parentSize.height / 4);
+        }
+        // set up layout
+        setLayout(new GridBagLayout());
+        GridBagConstraints constraints = new GridBagConstraints();
+        constraints.anchor = GridBagConstraints.WEST;
+        constraints.insets = new Insets(5, 5, 5, 5);
+
+        buttonImport.addActionListener(this);
+        buttonCancel.addActionListener(new ActionListener(){
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                ImportFromUrlDialog.this.setVisible(false);
+                ImportFromUrlDialog.this.dispose();
+            }});
+
+        // add components to the frame
+        constraints.gridx = 0;
+        constraints.gridy = 0;
+        add(labelURL, constraints);
+
+        constraints.gridx = 1;
+        constraints.fill = GridBagConstraints.HORIZONTAL;
+        constraints.weightx = 1.0;
+        constraints.gridwidth = 2;
+        fieldURL = addContextMenu(fieldURL);
+        add(fieldURL, constraints);
+
+        constraints.gridwidth = 1;
+        constraints.gridy = 2;
+        constraints.anchor = GridBagConstraints.CENTER;
+        add(buttonCancel, constraints);
+        constraints.gridx = 2;
+        constraints.gridy = 2;
+        constraints.anchor = GridBagConstraints.CENTER;
+        add(buttonImport, constraints);
+
+        setDefaultCloseOperation(DISPOSE_ON_CLOSE);
+        pack();
+        setVisible(true);
+    }
+
+    /* Action executed by import button. */
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        if (caller != null) {
+            String url = fieldURL.getText();
+            /* Calls a parsing task in a new thread. */
+            try {
+                caller.importOpenApiDefinition(new URI(url, false));
+            } catch (URIException ex) {
+                View.getSingleton().showWarningDialog(thisDialog, Constant.messages.getString(MESSAGE_PREFIX + "badurl"));
```

Comment:
This is only in the UI, and the user should get immediate feedback from the warning dialog. I think we should only log error for potential bugs in the code or things invoked via the API...